### DISCUSSION TITLE
ustack only print symbols of the first stacktrace for three times

### DISCUSSION THREAD

**Discussion by baiwfg2** [bpftrace/bpftrace#3614] [Q&A]
bpftrace 0.18 on aarch64

Hi, I've found some strange behaviors. Here's my bt file:
```
uprobe:/build/lib/libfoo.so:DbAdptFallocate
{
    printf("stacktrace:%s\n", ustack);
}
```
`DbAdptFallocate` is implemented in libfoo.so.

I know my program will call `DbAdptFallocate` three times. When I experiment it, it may output like this (a,b,c are just abstracted functions' name, they may be defined in different dynamic libraries):
```
stacktrace:
        DbAdptFallocate+0
	c
	b
        a
        main+84
        __libc_start_main+224
        0x400784

stacktrace:
        DbAdptFallocate+0
        c
        0xffffabc
        0xffffabd
stacktrace:
        0xffffabe
        0xffffabf
	0xffffabc
        0xffffabd 
```
That is , I often see the complete stack in the first stacktrace, see part stack in the second (sometimes all unresolved addresses), always see unresolved address in the third. Why is that ?

I try to experiment it on another x86_64 ubuntu host, bpftrace 0.18. It behaves much better, it seldom shows unresolved symbols.

Any help is appreciated.

**Comment by jordalgo** [comment]
Always my first question: are all the binaries/libraries compiled with frame pointers enabled?

**Reply by baiwfg2** [reply]
Yes. It's debug build and I double-check x29/x30 registers are saved at the beginning of functions.

**Comment by danobi** [comment]
Along with Jordan's question, please also test with bpftrace on master and check the problem still occurs. Appimage is easiest way to install from master: https://github.com/bpftrace/bpftrace/blob/master/INSTALL.md#appimage-install .

EDIT: unfortunately we don't have aarch64 appimages yet. So you might need to check out https://github.com/bpftrace/bpftrace/blob/master/docs/developers.md#building

**Reply by baiwfg2** [reply]
Thanks, I'm trying latest bpftrace release 0.21. But I met with some problem with nix build: https://github.com/bpftrace/bpftrace/discussions/3619
